reducing_utils_test.go

```diff
@@ -355,3 +355,54 @@ func Test_Nper(t *testing.T) {
 		})
 	}
 }
+
+func Test_rate(t *testing.T) {
+	type args struct {
+		pv           decimal.Decimal
+		fv           decimal.Decimal
+		pmt          decimal.Decimal
+		nper         int64
+		when         paymentperiod.Type
+		maxIter      int64
+		tolerance    decimal.Decimal
+		initialGuess decimal.Decimal
+	}
+	tests := []struct {
+		name string
+		args args
+		want decimal.Decimal
+	}{
+		{
+			name: "success", args: args{
+				pv:           decimal.NewFromInt(2000),
+				fv:           decimal.NewFromInt(-3000),
+				pmt:          decimal.NewFromInt(100),
+				nper:         4,
+				when:         paymentperiod.BEGINNING,
+				maxIter:      100,
+				tolerance:    decimal.NewFromFloat(1e-7),
+				initialGuess: decimal.NewFromFloat(0.1),
+			},
+			want: decimal.NewFromFloat(0.06106257989825202),
+		}, {
+			name: "success", args: args{
```

Comment:
add nan test case.

Reply:
so nan basically requires your validity to be false, so I've changed the test to handle that case as well